### PR TITLE
Handle WM_MOUSEHWHEEL for horizontal touchpad scrolling

### PR DESCRIPTION
**Pull request by mdonatas**
It turns out that `Panel` on which this editor is based only handles `WM_MOUSEWHEEL` which gives vertical scrolling but doesn't handle `WM_MOUSEHWHEEL` which is fired during horizontal scroll.
So I basically copy/pasted [this code](https://referencesource.microsoft.com/#System.Windows.Forms/winforms/Managed/System/WinForms/Control.cs,13550) used for vertical scrolling and hooked it up to vertical scroll events. Methods in `Win32Util` also came from there.

## Testing methodology
Manual testing with laptop-touchpad

## Before
https://user-images.githubusercontent.com/483659/196259300-682a9691-6371-4fb7-b649-430647e09245.mp4

## After
https://user-images.githubusercontent.com/483659/196259655-b21917cc-d3dd-424e-9a42-d1be80969565.mp4

p.s. I've also created a simple `Form` with a `Panel` and some oversized content in both x and y axises and confirmed that horizontal scrolling with touch-pad doesn't work
image

### CODE REVIEW

File: Project/Src/Gui/TextAreaControl.cs [hunk 1]
```diff
@@ -514,21 +541,29 @@ public void JumpTo(int line, int column)
 
         protected override void WndProc(ref Message m)
         {
-            if (m.Msg == 0x007B)
```
Comment (mdonatas) [inline Project/Src/Gui/TextAreaControl.cs:517]: This is just a conversion from `if` to `switch`

**Review by RussKie — APPROVED** [review]
🚀

**Comment by mdonatas** [comment]
Could I get a `hacktoberfest-accepted` label please 😊

**Comment by RussKie** [comment]
I decorated the repo with the topic as well.

**Pull request merged**